File: Angular-TodoApp/src/app/models/task.ts
export class Task {
    Id: number=0;
    Title: string='';
    Description: string='';
    IsCompleted: boolean=false;
    CreatedOn: Date;
    TaskDate: Date;
    CompletedOn: Date | undefined;
    constructor(args:any) {
        this.Id = args.id || args.Id || 0;
        this.Title = args.title || args.Title || '';
        this.Description = args.description || args.Description || '';
        this.IsCompleted = args.isCompleted || args.IsCompleted || false;
        this.CreatedOn = args.createdOn || args.CreatedOn || new Date();
        this.TaskDate = args.taskDate || args.TaskDate || new Date();
        this.CompletedOn = args.completedOn || args.CompletedOn;
    }
}

File: ToDoApp-Web API/ToDoApp.Models/Task.cs
namespace ToDoApp.Models
{
    public class Task : BaseEntity
    {
        public string Title { get; set; } = null!;
        public string? Description { get; set; }
        public int UserId { get; set; }
        public DateTime CreatedOn {  get; set; }
        public DateTime TaskDate { get; set; } = DateTime.MinValue;
        public bool IsCompleted { get; set; }
        public DateTime? CompletedOn { get; set; }   
    }
}